velociraptor/autoplotter/lines.py

```diff
@@ -290,8 +291,186 @@ def create_line(
+            # Y-axis range
+            x_down_list = x[below_y_within_x]
+            x_up_list = x[above_y_within_x]
+
+            # Use figure's data coordinates along the X axis and relative coordinates
+            # along the Y axis.
+            tform_x = blended_transform_factory(ax.transData, ax.transAxes)
+
+            # Draw arrows pointing downwards
+            for x_down in x_down_list:
+                ax.annotate(
+                    "",
+                    xytext=(x_down, arrow_length + distance_from_edge),
+                    textcoords=tform_x,
+                    xy=(x_down, distance_from_edge),
+                    xycoords=tform_x,
+                    arrowprops=dict(color=color, arrowstyle="->"),
+                )
+
+            # Draw arrows pointing upwards
+            for x_up in x_up_list:
+                ax.annotate(
+                    "",
+                    xytext=(x_up, 1.0 - arrow_length - distance_from_edge),
+                    textcoords=tform_x,
+                    xy=(x_up, 1.0 - distance_from_edge),
+                    xycoords=tform_x,
+                    arrowprops=dict(color=color, arrowstyle="->"),
+                )
+
+            # Next, find all data points that are outside the X-axis range and
+            # within Y-axis range
+            below_x_within_y = below_x_range * within_y_range
+            above_x_within_y = above_x_range * within_y_range
+
+            # Y coordinates of the data points whose X coordinates are outside the
+            # X-axis range
+            y_left_list = y[below_x_within_y]
+            y_right_list = y[above_x_within_y]
+
+            # Use figure's data coordinates along the Y axis and relative coordinates
+            # along the X axis.
+            tform_y = blended_transform_factory(ax.transAxes, ax.transData)
+
+            # Draw arrows pointing leftwards
+            for y_left in y_left_list:
+                ax.annotate(
+                    "",
+                    xytext=(arrow_length + distance_from_edge, y_left),
+                    textcoords=tform_y,
+                    xy=(distance_from_edge, y_left),
+                    xycoords=tform_y,
+                    arrowprops=dict(color=color, arrowstyle="->"),
+                )
+
+            # Draw arrows pointing rightwards
+            for y_right in y_right_list:
+                ax.annotate(
+                    "",
+                    xytext=(1.0 - arrow_length - distance_from_edge, y_right),
+                    textcoords=tform_y,
+                    xy=(1.0 - distance_from_edge, y_right),
+                    xycoords=tform_y,
+                    arrowprops=dict(color=color, arrowstyle="->"),
+                )
+
+            # Finally, handle the points that are both outside the X and Y axis range
+            outside_plot = (below_y_range + above_y_range) * (
+                below_x_range + above_x_range
+            )
```

Comment:
`logical_or` instead of +. Let's respect the fact that these are boolean arrays (even though they are implemented as integer arrays)

Reply:
Yes, I agree. That is better. This change is also in my latest commit.